types/valuerenderer/valuerenderer.go

```diff
@@ -0,0 +1,210 @@
+package valuerenderer
+
+import (
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+// ValueRenderer defines an interface to produce formated output for Int,Dec,Coin types as well as parse a string to Coin or Uint.
+type ValueRenderer interface {
+	Format(interface{}) (string, error)
+	Parse(string) (interface{}, error)
+}
+
+// DefaultValueRenderer defines a struct that implements ValueRenderer interface
+type DefaultValueRenderer struct {
+	// denomToMetadataMap represents a hashmap to lookup banktypes.Metadata by coin denom
+	denomToMetadataMap map[string]banktypes.Metadata
+}
+
+var _ ValueRenderer = &DefaultValueRenderer{}
+
+// TODO consider to move into valuerenderer_test.go
+// NewDefaultValueRenderer  initiates denomToMetadataMap field and returns DefaultValueRenderer struct
+func NewDefaultValueRenderer() DefaultValueRenderer {
```

Comment:
Since `NewDefaultValueRenderer` is used only in tests, does it make sense to move it into `valuerenderer-test` package? in this case i have to export `denomToMetadataMap` field.  Does it sound good ?

Reply:
```
type DefaultValueRenderer struct {
  denomQuerier func (ctx context.Context, denom string) bank.Metadata
}
```

if we're in teh state machine, pass in the keeper. If we're client-side, pass the client context